OEL-1247: Integration listing page. (#109)
openeuropa/oe_showcase: modules/oe_showcase_list_pages/README.md

# OpenEuropa Showcase List Pages

The scope of this module is to demo the OpenEuropa Library List pages.
This module uses [Open Europa Listing Page](https://github.com/openeuropa/oe_list_pages) and enables [Open Europa Whitelabel List Pages](https://github.com/openeuropa/oe_whitelabel). This module contains some facets exported and a dedicated index
and server for the search.

## Create a list page and setting up bundles and filters

> By default enables oe_list_pages, the news bundle as well as the title and publication date filters are available.

Reviewer: You changed this sentence, but it is still weird

Reviewer: We basically have two cases:
- If you are playing with showcase, all you need to do is enable `oe_showcase_list_pages` and create a custom list page for news.
- If you build your own website without showcase, you would enable `oe_list_pages` and `oe_whitelabel_list_pages`, then you would create search index and facets, then you would create your list page, or you would leave this to editors on production.

With the sentence above, I am not sure which case is being addressed here.
(talk to me in chat)

Reviewer: If you can think of an easy improvement, go ahead. Otherwise leave it be: I want to change the readme myself anyway. Perhaps in a follow-up issue.

Author: Improved it a bit

Reviewer: Still weird but let's address this in a follow-up.

### Create list pages

> * Go to add/oe_list_page.

Reviewer: ```suggestion
* Go to `node/add/oe_list_page`.
```

* Enter a title to the list page.
* Expand the "List Page" section in the right panel.
* Select the bundle on "Source bundle".
* Check "Override default exposed filters" and select the desired exposed filters you may as well set default values for these filters.

### To set up new bundles

> * Go to `config/search/search_api`.

Reviewer: ```suggestion
* Go to `config/search/search-api`.
```

* Add new index.
* Check Content on "Datasources".
* Select the desired bundle(s).
* Click on Save.
* Click on the tab fields to add the fields do be indexed.
* Click on save.

### Set up new filters
Filters are facets on search API module. To create new facets you should:

> * Go to "admin/config/search/facets".

Reviewer: Let's be consistent.
UI labels in quotes, url paths in backticks.
```suggestion
* Go to `admin/config/search/facets`.
```

# OpenEuropa Showcase List Pages

The scope of this module is to demo the OpenEuropa Library List pages.
This module uses [Open Europa Listing Page](https://github.com/openeuropa/oe_list_pages) and enables [Open Europa Whitelabel List Pages](https://github.com/openeuropa/oe_whitelabel). This module contains some facets exported and a dedicated index
and server for the search.

> ## Setting up bundles and filters

Reviewer: Title does not really describe what the section is about.

Reviewer: Still a bit weird, but let's resolve this one.

To create list pages enable the oe_list_pages module:
```
drush en oe_list_pages

> ```

Reviewer: Really?
In showcase, everything you need will be already enabled by default.
In a custom project, you want to enable `oe_list_pages` and `oe_whitelabel_list_pages`, if you use whitelabel theme.

Reviewer: Resolved, but I added a new comment.

By default when enabling oe_list_pages the news bundle as well as the title and publication date filters are available.

> ### Create list pages:

Reviewer: No colon (":") after hx elements.

* Go to add/oe_list_page.
* Enter a title to the list page.

> * Click on ```List Page``` link to collapse the available configuration.

Reviewer: No triple backticks for inline code snippets.
Also this is not really a code snippet, is it?
Just use regular quotes.
And "expand", not "collapse".
```suggestion
* Expand the "List Page" section in the right panel.
```

> * Select Content on the ```Source entity type```.

Reviewer: This is already the default value.

> * Select the bundle on ```Source bundle```.

Reviewer: Should be "News".

Author: depends cause if the user follows the instructions, could be custom.

Reviewer: Yeah, sorry. This was my earlier assumption that the readme is for users of showcase. But it is for users who create their own website.

* Check ```Override default exposed filters``` and select the desired exposed filters you may as well set default values for these filters.

> ### Add new bundles you should:

Reviewer: This phrase is weird. Sounds like Yoda talking :)

> * Go to ```config/search/search_api```.

Reviewer: ```suggestion
* Go to `config/search/search-api`.
```

Reviewer: I was going to say most of this stuff should be documented in oe_list_pages, not here. But perhaps it is ok to do it here, following the issue description.

Reviewer: Not fixed. But I created a new comment, so I will resolve this one.
